### PR TITLE
Add amqplib plugin for producer.

### PR DESCRIPTION
**Pull request by jmjoy**
- [x] Add amqplib plugin.
- [x] Refactor `try_get_sw_header` method.
- [x] Add integration tests.
- [x] Update documents.

### CODE REVIEW

**Comment by huangdijia** [comment]
Looking forward to supporting the swoole coroutine component as soon as possible.

File: src/component.rs [hunk 1]
```diff
@@ -24,3 +24,4 @@ pub const COMPONENT_PHP_MYSQLI_ID: i32 = 8004;
 pub const COMPONENT_PHP_PREDIS_ID: i32 = 8006;
 pub const COMPONENT_PHP_MEMCACHED_ID: i32 = 20;
 pub const COMPONENT_PHP_REDIS_ID: i32 = 7;
+pub const COMPONENT_RABBITMQ_PRODUCER_ID: i32 = 52;
```
Comment (wu-sheng) [inline src/component.rs:27]: So, only the producer side? No consumer?

Comment (jmjoy) [reply src/component.rs:27]: The consumer plan will be added later (PHP is not very suitable as a consumer).

File: README.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ SkyWalking PHP Agent requires SkyWalking 8.4+ and PHP 7.2+
   * [ ] [php-amqp](https://github.com/php-amqp/php-amqp)
   * [ ] [php-rdkafka](https://github.com/arnaud-lb/php-rdkafka)
   * [x] [predis](https://github.com/predis/predis)
+  * [x] [php-amqplib](https://github.com/php-amqplib/php-amqplib)
```
Comment (wu-sheng) [inline README.md:36]: Do you think we need to put `producer` here?
```suggestion
  * [x] [php-amqplib](https://github.com/php-amqplib/php-amqplib) for Message Queuing Producer.
```

Comment (wu-sheng) [reply README.md:36]: Also, you mentioned amqp here, but the component ID is `rabbitmq-producer`. Is only Rabbit MQ supported?

Comment (jmjoy) [reply README.md:36]: I will update the README and doc later.

Comment (jmjoy) [reply README.md:36]: Not only RabbitMQ supported, but I don't found the component ID about amqp protocol, the library `php-amqplib` is commonly used in message queues services which supports the ampq protocol.

Comment (wu-sheng) [reply README.md:36]: If you need a general type for `amqp`, you have to register it here, https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/component-libraries.yml

Also, you may need mapping to indicate it is a queue server(without a specific MQ type), or an icon for topology.

**Review by heyanlong — DISMISSED** [review]
LGTM

**Review by wu-sheng — APPROVED** [review]

**Comment by jmjoy** [comment]
> Looking forward to supporting the swoole coroutine component as soon as possible.

I don't use swoole frequently, but I found that swoole supports [SWOOLE_HOOK_ALL](https://wiki.swoole.com/#/runtime?id=swoole_hook_all), so can we reuse FPM ecological components?

**Pull request merged**